tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,967 @@
+
+                            old_kernel_launch = input_source[arguments[0]["start"]:arguments[-1]["end"]]
+                            new_kernel_launch = old_kernel_launch
+
+                            kernel_params = argument_strings[5:]
+                            for arg_idx, arg in enumerate(kernel_params):
+                                if arg_idx in argument_types:
+                                    arg = kernel_params[arg_idx].strip()
+                                    the_type = argument_types[arg_idx]
+                                    the_arg = arg.replace("\n", "").strip()
+                                    if the_type in ["int", "const int", "int64_t", "THCIndex_t *", "const int *", "ptrdiff_t", "long", "const int64_t*", "int64_t *", "double"]:
+                                        static_argument = "static_cast<%s>(%s)" % (the_type, the_arg)
+                                        static_argument = arg.replace(the_arg, static_argument)
+
+                                        # Update to static_cast
+                                        new_kernel_launch = re.sub(r'\b(%s)\b' %
+                                                                   arg, lambda x: static_argument, new_kernel_launch)
+
+                            # Add template type
+                            if "THCUNN" in filepath.split("/") and "generic" not in filepath.split("/"):
+                                kernel_name_with_template = kernel_name_with_template.replace("<real>", "<Dtype>")
+                            new_kernel_launch = re.sub(r'\b(%s)\b' % kernel_name,
+                                                       lambda x: kernel_name_with_template, new_kernel_launch)
+
+                            # Replace Launch
+                            new_output_source = new_output_source.replace(old_kernel_launch, new_kernel_launch)
+
+                    # Overwrite file contents
+                    fileobj.seek(0)
+                    fileobj.write(new_output_source)
+                    fileobj.truncate()
+                    fileobj.flush()
+
+                    # Flush to disk
+                    os.fsync(fileobj)
+
+
+def main():
+    """Example invocation
+
+    python hipify.py --project-directory /home/myproject/ --extensions cu cuh h cpp --output-directory /home/gains/
+    """
+
+    parser = argparse.ArgumentParser(
+        description="The Python Hipify Script.")
+
+    parser.add_argument(
+        '--project-directory',
+        type=str,
+        default=os.getcwd(),
+        help="The root of the project.",
+        required=True)
+
+    parser.add_argument(
+        '--show-detailed',
+        type=bool,
+        default=False,
+        help="Show detailed summary of the hipification process.",
+        required=False)
+
+    parser.add_argument(
+        '--extensions',
+        nargs='+',
+        default=["cu", "cuh", "c", "cpp", "h", "in", "hpp"],
```

Comment:
Don't have to do in this PR, but `in` is a bit of a dangerous extension to match over, since ANY file that is a template, e.g., `myrandom.txt.in`. It would be better if you could say `.h.in` and then adjust the code that reads the extensions to be able to handle extensions that have multiple components (if it doesn't support that already).

Reply:
Yes, let's do that later.